subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c

```diff
@@ -526,6 +542,19 @@ static int firmware_update_result(uint16_t obj_inst_id, uint16_t res_id, uint16_
 	return 0;
 }
 
+static void init_firmware_variables(void)
+{
+	if (fota_host) {
+		k_free(fota_host);
+		fota_host = NULL;
+	}
+	fota_path = NULL;
+	percent_downloaded = 0;
+	bytes_downloaded = 0;
+	ongoing_obj_id = UNUSED_OBJ_ID;
+	k_work_schedule(&pending_download_work, PENDING_DELAY);
```

Comment:
Missing error check for `k_work_schedule()`.
It can return an negative error code. You should handle that, and mark the FOTA as failed.